### ISSUE TITLE
Issue running tests through pytest with always_eager=True

### ISSUE THREAD

**Issue by ttt733** [coleifer/huey#363]
Hello there. My team is switching over to Huey from APScheduler, and so far things are looking great, but we've run into a complication getting our unit tests to pass. I'll share some trimmed code.

qhuey.py
```
from huey import RedisHuey
# We run into the issue whether blocking=True or blocking=False
huey = RedisHuey('testing', always_eager=True)
```

api.py
```
from qhuey import huey
#...
@huey.task()
def handle_event(event: Event, received_datetime: Optional[datetime] = None) -> None:
    print('Handling event...')
   # submit event to the store
```
api_test.py (Note: api.handle just wraps api.handle_event.)
```
def test_get_specific_events(api, capsys):
  # generate some events
  with capsys.disabled():
        api.handle(started_event)
        api.handle(suspended_event)
        api.handle(resumed_event)
        api.handle(canceled_event)

  events = api.get_events(vsi_id, None, None)
  assert len(events) == 4
  # some more assertions go here
  with capsys.disabled():
        print("all assertions passed")
```
If I run this, I see output like this:
```
metronome/tests/api_test.py Handling event...
```
It hangs here, but if I send a keyboard interrupt to the console running pytest...
```
metronome/tests/api_test.py Handling event...
^CHandling event...
Handling event...
Handling event...
all assertions passed
```

So the issue seems to be not that the jobs aren't being processed, but that it's simply hanging at some point waiting on something to happen. I'm having a hard time figuring out what might be the cause here. I saw that https://github.com/coleifer/huey/commit/2c465dc117e84b9d1119f8b06a5488f68a793e7d contained some code that fixed a somewhat similar situation for someone whose pipelines were only working for the first event, but I tried running this with 1.10.0 and 1.10.2 (which should have that) and saw the same behavior on both versions. Please let me know if you see something obvious that we're doing wrong or if you'd like more info.

Additional note: I made sure that the test passes with the `@huey.task()` decorator commented out.

**Comment by coleifer** [comment]
I have no idea what might be going on in your setup, in your functions, in "capsys.disabled()", etc, etc.

Could you identify where exactly in `huey` the code seems to be hanging? Have you tried setting a breakpoint and stepping into the code? As-is this is completely useless information you have provided me.

**Comment by ttt733** [comment]
`capsys.disabled()` is just a pytest command that stops it from eating
console logs and lets me show you the output I put towards the bottom of my
post. (Seeing the same behavior with it removed.) There's not much in the
way of setup code for our tests. Is there any particular part of the
setup/configs you're interested in? I'm not sure what additional info about
that I can provide - I'd probably need to get approval to post a more
significant amount of the code involved, and I'm not sure it'll help. I
could probably get that on Monday for you, but as I said, the tests work
when the function being tested is not decorated with huey.task(), so I'm
thinking it has more to do with Huey or the way we're using it than the
server/test logic.

Running pytest on our project with a debugger is a somewhat finnicky
process, but if you think that's an avenue I should pursue, I can put some
time in to getting that running on Monday and see where it's hanging.
(Worst case, I think I could figure out where to hack some print statements
in, or try writing a more minimal breaking case in a new project.) I just
thought that, since it works without the scheduling layer (which, from my
understanding, should be pretty thin with `always_eager=True` set) I should
check and see if there was something obvious we weren't doing correctly
with it or if there might be another issue here.

If there are any watchers who happen to be looking and are successfully
running tasks through pytest in a similar fashion (i.e. without pipelines,)
using the always_eager scheduler and Huey>=1.10.0, I'd appreciate seeing
some snippets of your implementation.

**Comment by coleifer** [comment]
> Running pytest on our project with a debugger is a somewhat finnicky
process

And you don't suspect that this might be the problem?

There's no use in speculating until you can point me to where in `huey` things are getting locked up. Set a breakpoint. Use `pdb.set_trace()` or something to step into the code. I'll gladly reopen if you can provide something concrete for me to go on.

**Comment by coleifer** [comment]
I'm still not clear on how this is a Huey issue. Are you sure "args" is a single `Event` instance? Based on what you've given me, here is the simplest example I can think of showing that huey works correctly:

```python
from huey import RedisHuey

huey = RedisHuey(always_eager=True)

class Event(object):
    def __init__(self, name):
        self.name = name
    def __repr__(self):
        return '<Event: name=%s>' % self.name
    def __getitem__(self, item):
        return self.__dict__.get(item)
    get = __getitem__

@huey.task()
def event_task(event):
    print('task received', event.name)
    return event.name

events = [Event(name='name-%s' % i) for i in range(10)]
for event in events:
    event_task(event)
```

Seems to work fine -- prints 'task received name-#' 10x.

**Comment by ttt733** [comment]
Yeah, this is strange. I tried it with your minimal case in the same virutalenv, worked out to run through pytest and everything, and it works. Debugging got me this:
```
(Pdb) ll
874         def execute(self):
875             args, kwargs = self.data or ((), {})
876             if retries_as_argument:
877                 kwargs['retries'] = self.retries
878             if include_task:
879                 kwargs['task'] = self
880  ->         return func(*args, **kwargs)
(Pdb) p self.data
((<Event: name={'name': 'event1'}>,), {})
(Pdb) p args
(<Event: name={'name': 'event1'}>,)
```
As shown above, when I printed args on my project at the same point , I got this:
```
(Pdb) p args
<Event{'id': None, 'uuid': None, ...}>
```
So it's not as simple as I hoped. I'll do some more digging and see why it's going in as an event in my project.

**Comment by ttt733** [comment]
Ugh, this can stay closed. Someone wrote a handle_event.schedule() call that I didn't know about into our code with `args=(Event.create(**kwargs))` instead of `args=(Event.create(**kwargs),)`, so python just ignored the parentheses and didn't put it in as a tuple. Sorry about the hassle.
